src/Symfony/Bundle/FrameworkBundle/EventListener/NoRoutingConfigurationSubscriber.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Routing\Exception\NoConfigurationException;
+
+/**
+ * Show welcome message if no routing configuration is available.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class NoRoutingConfigurationSubscriber implements EventSubscriberInterface
+{
+    private $projectDir;
+
+    public function __construct($projectDir)
+    {
+        $this->projectDir = $projectDir;
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        $exception = $event->getException();
+        if ($exception instanceof NotFoundHttpException && $exception->getPrevious() instanceof NoConfigurationException) {
+            $event->setResponse(new Response($this->getContent()));
```

Comment:
Still the final status code is `404`, it's OK for you?

Reply:
Yes, should be 404.